ngraph_bridge/ngraph_encapsulate_op.cc

```diff
@@ -53,1019 +57,638 @@ namespace ng = ngraph;
-        NGRAPH_VLOG(1) << "NGRAPH_TF_MEM_PROFILE:  OP_ID: " << my_instance_id
-                       << " Step_ID: " << ctx->step_id()
-                       << " Cluster: " << ctx->op_kernel().name()
-                       << " Input Tensors freed: "
-                       << input_tensors_bytes_free / (1024 * 1024) << " MB"
-                       << " Output Tensors freed: "
-                       << output_tensors_bytes_free / (1024 * 1024) << " MB";
-      }  // cache eviction if cache size greater than cache depth
-
-      BackendManager::LockBackend(m_op_backend_name);
-
-      ngraph::Event event_compile("Compile nGraph", name(), "");
-      try {
-        ng_exec = op_backend->compile(ng_function);
-
-      } catch (const std::exception& exp) {
-        BackendManager::UnlockBackend(m_op_backend_name);
-        NgraphSerialize(
-            "tf_function_error_" + ctx->op_kernel().name() + ".json",
-            ng_function);
-        return errors::Internal("Caught exception while compiling op_backend: ",
-                                exp.what(), "\n");
-      } catch (...) {
-        BackendManager::UnlockBackend(m_op_backend_name);
-        NgraphSerialize(
-            "tf_function_error_" + ctx->op_kernel().name() + ".json",
-            ng_function);
-        return errors::Internal("Error in compiling op_backend\n");
-      }
-      BackendManager::UnlockBackend(m_op_backend_name);
-      event_compile.Stop();
-      ngraph::Event::write_trace(event_compile);
-
-      m_ng_exec_map[signature] = ng_exec;
-      // caching ng_function to serialize to ngraph if needed
-      m_ng_function_map[ng_exec] = ng_function;
-
-      m_lru.push_front(signature);
-      // Memory after
-      MemoryProfile(vm, rss);
-      auto delta_vm_mem = vm - vm0;
-      auto delta_res_mem = rss - rss0;
-      NGRAPH_VLOG(1) << "NGRAPH_TF_CACHE_PROFILE: OP_ID: " << my_instance_id
-                     << " Step_ID: " << ctx->step_id()
-                     << " Cache length: " << m_ng_exec_map.size()
-                     << "  Cluster: " << ctx->op_kernel().name()
-                     << " Delta VM: " << delta_vm_mem
-                     << "  Delta RSS: " << delta_res_mem
-                     << "  Function size: " << function_size
-                     << " KB Total RSS: " << rss / (1024 * 1024) << " GB "
-                     << " VM: " << vm / (1024 * 1024) << " GB" << endl;
-    }  // end of input signature not found in m_ng_exec_map
-    else {
-      // Found the input signature in m_ng_exec_map, use the cached executable
-      // Update the m_lru
-      if (signature != m_lru.front()) {
-        m_lru.remove(signature);
-        m_lru.push_front(signature);
-      }
-      ng_exec = it->second;
+// Dealing with the output from Variable nodes here
+#if defined(NGRAPH_TF_ENABLE_VARIABLES_AND_OPTIMIZERS)
+  NGRAPH_VLOG(4) << "NGraphEncapsulateOp::Compute getting output variables "
+                    "from resource manager "
```

Comment:
This block needs to be moved before
```
event_alloc_output.Stop();
```
That is the time for this loop needs to be attributed to the allocation of outputs.
Same thing for the block for inputs. Need to be moved next to NgAllocateInputs() and inside the block for event_alloc_input

Reply:
Just noticed you have separate blocks. So ignore this comment. The 2 #ifdefined are next to each other can be made one